app/main/subsystems/appState.ts

```diff
@@ -0,0 +1,33 @@
+import { AppStateManager } from "app/main/appState"
+import { dummyStop, Lifecycle } from "app/main/lifecycle"
+import { SubSystems } from "app/main/system"
+
+/**
+ * App state subsystem.
+ * This simply initializes and returns an app state manager.
+ */
+export class AppStateSubSystem implements Lifecycle<AppStateManager, SubSystems> {
+
+  /**
+   * A reference to the app state manager object itself.
+   */
+  private manager: AppStateManager
```

Comment:
I got confused by this because I was expecting TypeScript to complain since `manager` is uninitialized, in which case the type should be `AppStateManager | null`. Maybe we should enable also [strictPropertyInitialization](https://blog.mariusschulz.com/2018/05/20/typescript-2-7-strict-property-initialization). Actually enabling this property teaches us something, what we are doing here is pure initialization, which is a bit different than lifecycle, there's no state that needs to start or stop, so this is identical to the [pure json component](https://github.com/witnet/sheikah/blob/master/app/main/subsystems/json.ts#L5)

Reply:
Now that you mention it... would to the following `pureSubsystem` be 100% equivalent to an instance of my verbose `AppStateSubSystem` class?
```ts
const appStateSubSystem = pureSubsystem(() => new AppStateManager())
```
If so, I support going the `pureSubsystem` way to reduce code repetition.

Reply:
yes it is hahaha, the idea of the lifecycle is for components where initialization _is not enough_ (connecting to a db, reading some config remotely, etc), components that only need initialization is what I call "pure component or subsystem" :grin: